gif-convert

```diff
@@ -9,7 +9,7 @@
 start_at=0
 end_at="$(ffmpeg -i "$1" 2>&1 | grep "Duration" | cut -d ' ' -f 4 | sed s/,//)"
 fps="$(ffmpeg -i "$1" 2>&1 | grep "fps" | grep -o "[0-9.]*.fps" | grep -o "[0-9.]*")"
-scale="$(ffmpeg -i "$1" 2>&1 | grep "fps" | grep -o "[0-9]*.x[0-9]*")"
+scale="$(ffmpeg -i "$1" 2>&1 | grep "fps" | grep -Po "\d{3,5}x\d{3,5}")"
```

Comment:
Can you briefly explain what does this change does? 
The previous regex looks like a superset of the new regex to me, and that should work fine. Also, I am unable to learn about the `-P` flag.

Reply:
Sorry for the late reply. So this regex `"\d{3,5}x\d{3,5}"` means to grep at least 3 digit and 5 for maximal. Because I ever convert video and it outputs the resolution and something like 0x1230348. I use the new regex to ensure that the output is just the video resolution because video resolution at least has 3 digit. Ah and I also get that new regex from [this](https://superuser.com/a/1303373)